app/code/Magento/Eav/Model/ResourceModel/ReadHandler.php

```diff
@@ -112,36 +145,53 @@ public function execute($entityType, $entityData, $arguments = [])
         if (count($attributeTables)) {
             $attributeTables = array_keys($attributeTables);
             foreach ($attributeTables as $attributeTable) {
+                $from = ['value' => 't.value', 'attribute_id' => 't.attribute_id'];
+                if ($loadAllScopes) {
+                    $from['store_id'] = 't.store_id';
+                }
                 $select = $connection->select()
                     ->from(
                         ['t' => $attributeTable],
-                        ['value' => 't.value', 'attribute_id' => 't.attribute_id']
+                        $from
                     )
                     ->where($metadata->getLinkField() . ' = ?', $entityData[$metadata->getLinkField()]);
-                foreach ($context as $scope) {
-                    //TODO: if (in table exists context field)
-                    $select->where(
-                        $metadata->getEntityConnection()->quoteIdentifier($scope->getIdentifier()) . ' IN (?)',
-                        $this->getContextVariables($scope)
-                    )->order('t.' . $scope->getIdentifier() . ' DESC');
+                // TODO: Clean up this code
+                if (!$loadAllScopes) {
+                    foreach ($context as $scope) {
+                        //TODO: if (in table exists context field)
+                        $select->where(
+                            $metadata->getEntityConnection()->quoteIdentifier($scope->getIdentifier()) . ' IN (?)',
+                            $this->getContextVariables($scope)
+                        )->order('t.' . $scope->getIdentifier() . ' DESC');
+                    }
                 }
                 $selects[] = $select;
             }
             $unionSelect = new \Magento\Framework\DB\Sql\UnionExpression(
```

Comment:
If the `$loadAllScopes` variable is set to `false`, the resulting query looks like this:

```sql
(SELECT `t`.`value`, `t`.`attribute_id` FROM `catalog_category_entity_varchar` AS `t` WHERE (row_id = '15') AND (`store_id` IN (1, 0)) ORDER BY `t`.`store_id` DESC)
UNION ALL (SELECT `t`.`value`, `t`.`attribute_id` FROM `catalog_category_entity_int` AS `t` WHERE (row_id = '15') AND (`store_id` IN (1, 0)) ORDER BY `t`.`store_id` DESC)
UNION ALL (SELECT `t`.`value`, `t`.`attribute_id` FROM `catalog_category_entity_text` AS `t` WHERE (row_id = '15') AND (`store_id` IN (1, 0)) ORDER BY `t`.`store_id` DESC)
UNION ALL (SELECT `t`.`value`, `t`.`attribute_id` FROM `catalog_category_entity_datetime` AS `t` WHERE (row_id = '15') AND (`store_id` IN (1, 0)) ORDER BY `t`.`store_id` DESC)
UNION ALL (SELECT `t`.`value`, `t`.`attribute_id` FROM `catalog_category_entity_decimal` AS `t` WHERE (row_id = '15') AND (`store_id` IN (1, 0)) ORDER BY `t`.`store_id` DESC);
```

And the results look like this:

|value|attribute_id|
|-----|------------|
|Jackets|45|
|global|49|
|PRODUCTS|52|
|name|68|
|jackets-men|122|
|men/tops-men/jackets-men|123|
|0|197|
|1|46|
|1|54|
|1|69|
|0|70|
|1|71|
|<p>Global value</p>|47|
|global|50|
|name|67|
|100.0000|72|


If the `$loadAllScopes` variable is set to `true`, the resulting query looks like this:

```sql
(SELECT `t`.`value`, `t`.`attribute_id`, `t`.`store_id` FROM `catalog_category_entity_varchar` AS `t` WHERE (row_id = '15') ORDER BY `t`.`store_id` DESC)
UNION ALL (SELECT `t`.`value`, `t`.`attribute_id`, `t`.`store_id` FROM `catalog_category_entity_int` AS `t` WHERE (row_id = '15') ORDER BY `t`.`store_id` DESC)
UNION ALL (SELECT `t`.`value`, `t`.`attribute_id`, `t`.`store_id` FROM `catalog_category_entity_text` AS `t` WHERE (row_id = '15') ORDER BY `t`.`store_id` DESC)
UNION ALL (SELECT `t`.`value`, `t`.`attribute_id`, `t`.`store_id` FROM `catalog_category_entity_datetime` AS `t` WHERE (row_id = '15') ORDER BY `t`.`store_id` DESC)
UNION ALL (SELECT `t`.`value`, `t`.`attribute_id`, `t`.`store_id` FROM `catalog_category_entity_decimal` AS `t` WHERE (row_id = '15') ORDER BY `t`.`store_id` DESC);
```

And the results look like this:

|value|attribute_id|store_id|
|-----|------------|--------|
|Jackets|45|0|
|Jackets (website 1 store2)|45|4|
|homepage_category_endurance.jpg|48|4|
|global|49|0|
|Jackets (website 1 store2)|49|4|
|PRODUCTS|52|0|
|PRODUCTS|52|4|
|name|68|0|
|jackets-men|122|0|
|jackets-website-1-store2|122|4|
|men/tops-men/jackets-men|123|0|
|men/tops-men/jackets-website-1-store2|123|4|
|0|197|0|
|0|197|4|
|1|46|0|
|1|46|4|
|3|53|4|
|1|54|0|
|1|69|0|
|0|69|4|
|0|70|0|
|0|70|4|
|1|71|0|
|1|71|4|
|<p>Global value</p>|47|0|
|global|50|0|
|name|67|0|
|<p>Jackets (website 1 store2)</p>|47|4|
|Jackets (website 1 store2)|50|4|
|position,name|67|4|
|100.0000|72|0|
|100.0000|72|4|